src/coreclr/src/vm/eepolicy.cpp

```diff
@@ -1056,7 +1149,37 @@ void DECLSPEC_NORETURN EEPolicy::HandleFatalStackOverflow(EXCEPTION_POINTERS *pE
 
     STRESS_LOG0(LF_EH, LL_INFO100, "In EEPolicy::HandleFatalStackOverflow\n");
 
-    DisplayStackOverflowException();
+    FrameWithCookie<FaultingExceptionFrame> fef;
+#if defined(FEATURE_EH_FUNCLETS)
+    *((&fef)->GetGSCookiePtr()) = GetProcessGSCookie();
+#endif // FEATURE_EH_FUNCLETS
+    if (pExceptionInfo && pExceptionInfo->ContextRecord)
+    {
+        GCX_COOP();
+        AdjustContextForJITHelpers(pExceptionInfo->ExceptionRecord, pExceptionInfo->ContextRecord);
+        fef.InitAndLink(pExceptionInfo->ContextRecord);
+    }
+
+    static CLREvent* g_stackOverflowCallStackLoggedEvent = NULL;
+
+    // Dump stack trace only for the first thread failing with stack overflow to prevent mixing
+    // multiple stack traces together.
+    CLREvent* event = new CLREvent();
```

Comment:
This can throw exceptions, etc. It may be better to poll with `Sleep(50)` here.

Reply:
Makes sense, I've added a new commit replacing the event by polling as you've suggested.